### ISSUE TITLE
Sample ID SAMN* regex bug

### ISSUE THREAD

**Issue by martinghunt** [enasequence/enaBrowserTools#24]
The regex on line 106 of `utils.py` in the python3 code:
```
sample_pattern_1 = re.compile('^SAM[ND][0-9]{7}$')
```
doesn't match everything, eg SAMN03649073 because it has 8 digits. It makes `enaGroupGet` throw an error:
```
enaGroupGet -f fastq SAMN03649073
ERROR: Invalid accession. Only sample and study/project accessions or NCBI tax ID supported
```
I don't know what the spec is for these accessions, otherwise I'd put in a pull request. Python2 code also has the same regex.

**Comment by nicsilvester** [comment]
Thanks.  You're correct.  NCBI and DDBJ bio sample accessions have 8 digits, EBI ones start at seven. Thanks for the catch. I'll add it into the update I'm currently working on

**Comment by martinghunt** [comment]
Thanks!

**Comment by nicsilvester** [comment]
Fixed in v1.4 release
